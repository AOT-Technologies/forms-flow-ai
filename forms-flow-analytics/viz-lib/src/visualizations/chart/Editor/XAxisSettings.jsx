import React from "react";
import { Section, Switch } from "@/components/visualizations/editor";
import { EditorPropTypes } from "@/visualizations/prop-types";

import AxisSettings from "./AxisSettings";

export default function XAxisSettings({ options, onOptionsChange }) {
  return (
    <React.Fragment>
      <AxisSettings
        id="XAxis"
        features={{ autoDetectType: true }}
        options={options.xAxis}
        onChange={xAxis => onOptionsChange({ xAxis })}
      />

      <Section>
        <Switch
          data-test="Chart.XAxis.Sort"
          defaultChecked={options.sortX}
          onChange={sortX => onOptionsChange({ sortX })}>
          Sort Values
        </Switch>
      </Section>

      <Section>
        <Switch
          data-test="Chart.XAxis.Reverse"
          defaultChecked={options.reverseX}
          onChange={reverseX => onOptionsChange({ reverseX })}>
          Reverse Order
        </Switch>
      </Section>

      <Section>
        <Switch
          data-test="Chart.XAxis.ShowLabels"
          defaultChecked={options.xAxis.labels.enabled}
          onChange={enabled => onOptionsChange({ xAxis: { labels: { enabled } } })}>
          Show Labels
        </Switch>
      </Section>
    </React.Fragment>
  );
}

XAxisSettings.propTypes = EditorPropTypes;
