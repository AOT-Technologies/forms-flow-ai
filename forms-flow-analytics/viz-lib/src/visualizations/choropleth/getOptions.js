import { merge } from "lodash";
import ColorPalette from "./ColorPalette";

const DEFAULT_OPTIONS = {
  mapType: "countries",
  countryCodeColumn: "",
  countryCodeType: "iso_a3",
  valueColumn: "",
  clusteringMode: "e",
  steps: 5,
  valueFormat: "0,0.00",
  noValuePlaceholder: "N/A",
  colors: {
    min: ColorPalette["Light Blue"],
    max: ColorPalette["Dark Blue"],
    background: ColorPalette.White,
    borders: ColorPalette.White,
    noValue: ColorPalette["Light Gray"],
  },
  legend: {
    visible: true,
    position: "bottom-left",
    alignText: "right",
  },
  tooltip: {
    enabled: true,
    template: "<b>{{ @@name }}</b>: {{ @@value }}",
  },
  popup: {
    enabled: true,
    template: "Country: <b>{{ @@name_long }} ({{ @@iso_a2 }})</b>\n<br>\nValue: <b>{{ @@value }}</b>",
  },
};

export default function getOptions(options) {
  return merge({}, DEFAULT_OPTIONS, options);
}
