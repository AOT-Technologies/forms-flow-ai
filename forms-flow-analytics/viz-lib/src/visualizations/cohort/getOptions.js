import { merge } from "lodash";
import ColorPalette from "@/visualizations/ColorPalette";

const DEFAULT_OPTIONS = {
  timeInterval: "daily",
  mode: "diagonal",
  dateColumn: "date",
  stageColumn: "day_number",
  totalColumn: "total",
  valueColumn: "value",

  showTooltips: true,
  percentValues: true,

  timeColumnTitle: "Time",
  peopleColumnTitle: "Users",
  stageColumnTitle: "{{ @ }}",

  numberFormat: "0,0[.]00",
  percentFormat: "0.00%",
  noValuePlaceholder: "-",

  colors: {
    min: "#ffffff",
    max: ColorPalette["Dark Blue"],
    steps: 7,
  },
};

export default function getOptions(options) {
  return merge({}, DEFAULT_OPTIONS, options);
}
