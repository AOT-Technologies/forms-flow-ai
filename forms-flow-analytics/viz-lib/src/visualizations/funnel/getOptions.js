import { isFinite, map, merge, includes } from "lodash";

const DEFAULT_OPTIONS = {
  stepCol: { colName: null, displayAs: "Steps" },
  valueCol: { colName: null, displayAs: "Value" },
  autoSort: true,
  sortKeyCol: { colName: null, reverse: false },
  itemsLimit: 100,
  percentValuesRange: { min: 0.01, max: 1000.0 },
  numberFormat: "0,0[.]00",
  percentFormat: "0[.]00%",
};

export default function getOptions(options, { columns }) {
  options = merge({}, DEFAULT_OPTIONS, options);

  // Validate
  const availableColumns = map(columns, c => c.name);
  if (!includes(availableColumns, options.stepCol.colName)) {
    options.stepCol.colName = null;
  }
  if (!includes(availableColumns, options.valueCol.colName)) {
    options.valueCol.colName = null;
  }
  if (!includes(availableColumns, options.sortKeyCol.colName)) {
    options.sortKeyCol.colName = null;
  }

  if (!isFinite(options.itemsLimit)) {
    options.itemsLimit = DEFAULT_OPTIONS.itemsLimit;
  }
  if (options.itemsLimit < 2) {
    options.itemsLimit = 2;
  }

  if (options.autoSort) {
    options.sortKeyCol.colName = options.valueCol.colName;
    options.sortKeyCol.reverse = true;
  }

  return options;
}
