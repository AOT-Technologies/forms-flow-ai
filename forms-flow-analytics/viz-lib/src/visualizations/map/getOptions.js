import { merge } from "lodash";

const DEFAULT_OPTIONS = {
  latColName: "lat",
  lonColName: "lon",
  classify: null,
  groups: {},
  mapTileUrl: "//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
  clusterMarkers: true,
  customizeMarkers: false,
  iconShape: "marker",
  iconFont: "circle",
  foregroundColor: "#ffffff",
  backgroundColor: "#356AFF",
  borderColor: "#356AFF",
  bounds: null,
  tooltip: {
    enabled: false,
    template: "",
  },
  popup: {
    enabled: true,
    template: "",
  },
};

export default function getOptions(options) {
  options = merge({}, DEFAULT_OPTIONS, options);
  options.mapTileUrl = options.mapTileUrl || DEFAULT_OPTIONS.mapTileUrl;

  // Backward compatibility
  if (options.classify === "none") {
    options.classify = null;
  }

  return options;
}
