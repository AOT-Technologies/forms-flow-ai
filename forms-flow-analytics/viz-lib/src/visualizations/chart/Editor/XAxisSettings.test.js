import React from "react";
import enzyme from "enzyme";

import getOptions from "../getOptions";
import XAxisSettings from "./XAxisSettings";

function findByTestID(wrapper, testId) {
  return wrapper.find(`[data-test="${testId}"]`);
}

function mount(options, done) {
  options = getOptions(options);
  return enzyme.mount(
    <XAxisSettings
      visualizationName="Test"
      data={{ columns: [], rows: [] }}
      options={options}
      onOptionsChange={changedOptions => {
        expect(changedOptions).toMatchSnapshot();
        done();
      }}
    />
  );
}

describe("Visualizations -> Chart -> Editor -> X-Axis Settings", () => {
  test("Changes axis type", done => {
    const el = mount(
      {
        globalSeriesType: "column",
        xAxis: { type: "-", labels: { enabled: true } },
      },
      done
    );

    findByTestID(el, "Chart.XAxis.Type")
      .last()
      .simulate("click");
    findByTestID(el, "Chart.XAxis.Type.Linear")
      .last()
      .simulate("click");
  });

  test("Changes axis name", done => {
    const el = mount(
      {
        globalSeriesType: "column",
        xAxis: { type: "-", labels: { enabled: true } },
      },
      done
    );

    findByTestID(el, "Chart.XAxis.Name")
      .last()
      .simulate("change", { target: { value: "test" } });
  });

  test("Sets Show Labels option", done => {
    const el = mount(
      {
        globalSeriesType: "column",
        xAxis: { type: "-", labels: { enabled: false } },
      },
      done
    );

    findByTestID(el, "Chart.XAxis.ShowLabels")
      .last()
      .simulate("click");
  });

  test("Sets Sort X Values option", done => {
    const el = mount(
      {
        globalSeriesType: "column",
        sortX: false,
      },
      done
    );

    findByTestID(el, "Chart.XAxis.Sort")
      .last()
      .simulate("click");
  });

  test("Sets Reverse X Values option", done => {
    const el = mount(
      {
        globalSeriesType: "column",
        reverseX: false,
      },
      done
    );

    findByTestID(el, "Chart.XAxis.Reverse")
      .last()
      .simulate("click");
  });
});
