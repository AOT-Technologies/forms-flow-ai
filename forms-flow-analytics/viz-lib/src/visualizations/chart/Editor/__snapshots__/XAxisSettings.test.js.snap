// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualizations -> Chart -> Editor -> X-Axis Settings Changes axis name 1`] = `
Object {
  "xAxis": Object {
    "labels": Object {
      "enabled": true,
    },
    "title": Object {
      "text": "test",
    },
    "type": "-",
  },
}
`;

exports[`Visualizations -> Chart -> Editor -> X-Axis Settings Changes axis type 1`] = `
Object {
  "xAxis": Object {
    "labels": Object {
      "enabled": true,
    },
    "type": "linear",
  },
}
`;

exports[`Visualizations -> Chart -> Editor -> X-Axis Settings Sets Reverse X Values option 1`] = `
Object {
  "reverseX": true,
}
`;

exports[`Visualizations -> Chart -> Editor -> X-Axis Settings Sets Show Labels option 1`] = `
Object {
  "xAxis": Object {
    "labels": Object {
      "enabled": true,
    },
  },
}
`;

exports[`Visualizations -> Chart -> Editor -> X-Axis Settings Sets Sort X Values option 1`] = `
Object {
  "sortX": true,
}
`;
